### PR TITLE
feat: add vercel chat example

### PR DESCRIPTION
**Pull request by thucpn**

### CODE REVIEW

File: examples/vercel/tool.ts [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+  const queryEngine = index.asQueryEngine();
+  return {
+    description: description ?? "Get information about your documents.",
+    parameters: z.object({
+      query: z
+        .string()
+        .describe("The query to get information about your documents."),
+    }),
+    execute: async ({ query }) => {
+      const result = await queryEngine?.query({ query });
```
Comment (marcusschiesser) [inline examples/vercel/tool.ts:21]: Missing cases when result is missing

File: examples/vercel/tool.ts [hunk 2]
```diff
@@ -0,0 +1,25 @@
+import { CoreTool } from "ai";
+import { LlamaCloudIndex, VectorStoreIndex } from "llamaindex";
+import { z } from "zod";
+
+export async function llamaindex({
```
Comment (marcusschiesser) [inline examples/vercel/tool.ts:5]: How about moving to a @llamaindex/vercel package in providers folder?

Comment (marcusschiesser) [reply examples/vercel/tool.ts:5]: Use ai as peer dependency

File: examples/vercel/llamacloud.ts [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+  const path = "node_modules/llamaindex/examples/abramov.txt";
+  const essay = await fs.readFile(path, "utf-8");
+  const document = new Document({ text: essay, id_: path });
+
+  const index = await LlamaCloudIndex.fromDocuments({
+    documents: [document],
+    name: "test-pipeline",
+    projectName: "Default",
+    apiKey: process.env.LLAMA_CLOUD_API_KEY,
+    baseUrl: process.env.LLAMA_CLOUD_BASE_URL,
```
Comment (marcusschiesser) [inline examples/vercel/llamacloud.ts:17]: Not needed

File: examples/vercel/llamacloud.ts [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+  });
+  const queryTool = await llamaindex({ index });
```
Comment (marcusschiesser) [inline examples/vercel/llamacloud.ts:19]: Add description and add a comment that it's optional

**Review by marcusschiesser — APPROVED** [review]
LGTM

File: packages/providers/vercel/src/tool.ts [hunk 5]
```diff
@@ -0,0 +1,29 @@
+import type { BaseQueryEngine } from "@llamaindex/core/query-engine";
+import type { CoreTool } from "ai";
+import { z } from "zod";
+
+interface DatasourceIndex {
+  asQueryEngine: () => BaseQueryEngine;
+}
+
+export async function llamaindex({
```
Comment (marcusschiesser) [inline packages/providers/vercel/src/tool.ts:9]: @thucpn why has that to by async?

**Review by himself65 — APPROVED** [review]

**Pull request merged**